ADotNet/Models/Pipelines/GithubPipelines/DotNets/TagJob.cs

```diff
@@ -0,0 +1,28 @@
+﻿// ---------------------------------------------------------------------------
+// Copyright (c) Hassan Habib & Shri Humrudha Jagathisun All rights reserved.
+// Licensed under the MIT License.
+// See License.txt in the project root for license information.
+// ---------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using ADotNet.Models.Pipelines.GithubPipelines.DotNets.Tasks;
+using YamlDotNet.Serialization;
+
+namespace ADotNet.Models.Pipelines.GithubPipelines.DotNets
+{
+    public partial class TagJob
+    {
+        [YamlMember(Alias = "if")]
+        public string TagTrait = "github.event_name == 'pull_request' && github.event.action == 'closed'" +
+                " && github.event.pull_request.merged && startsWith(github.event.pull_request.title, 'RELEASES:')" + "" +
+                    " && contains(github.event.pull_request.labels.*.name, 'RELEASES')";
+
+        [YamlMember(Alias = "needs")]
+        public string[] Needs = { "build", "test" };
```

Comment:
I dont think we should hardcode the values "build" & "test".  This libary should give the consumer the flexibility to build it up as needed.

```suggestion
        [YamlMember(Alias = "needs", DefaultValuesHandling = DefaultValuesHandling.OmitDefaults)]
        public string[] Needs = { get; set; };
```